pkg/envoy/server.go

```diff
@@ -453,25 +482,37 @@ func getDirectionNetworkPolicy(l4Policy policy.L4PolicyMap, policyEnforced bool,
 			Rules:    make([]*cilium.PortNetworkPolicyRule, 0, len(l4.L7RulesPerEp)),
 		}
 
+		var allowAll bool
 		for sel, l7 := range l4.L7RulesPerEp {
-			rule := getPortNetworkPolicyRule(sel, l4.L7Parser, l7, labelsMap, deniedIdentities)
+			rule := getPortNetworkPolicyRule(sel, l4.L7Parser, l7, labelsMap, allowedIdentities, deniedIdentities)
 			if rule != nil {
 				if len(rule.RemotePolicies) == 0 && rule.L7Rules == nil {
 					// Got an allow-all rule, which would short-circuit all of
 					// the other rules. Just set no rules, which has the same
 					// effect of allowing all.
+					allowAll = true
 					pnp.Rules = nil
 					break
 				}
 
 				pnp.Rules = append(pnp.Rules, rule)
 			}
 		}
+
+		// No rule would match any traffic to/from the port. Just ignore the whole port.
+		if !allowAll && len(pnp.Rules) == 0 {
+			continue
+		}
+
```

Comment:
Ignoring the port will block all traffic on that port?

Reply:
Oops, I thought this was the rule that was supposed to allow all traffic for the port, I lost the context here. Maybe update the comment to make it clearer?

Reply:
Sure, I will update the comment to:
```
 		// No rule for this port matches any remote identity.
		// This means that no traffic was explicitly allowed at L4+ for this
		// port (traffic may still be allowed at L3-only).
		// In this case, just don't generate any PortNetworkPolicy for this
		// port.
```